### PR TITLE
Relationship between Adopted Resources with other aws-prometheusservice-controller components.

### PR DESCRIPTION
**Pull request by RipulHandoo**
**Notes for Reviewers**

This PR fixes #
1.Relationship between Adopted Resources with other aws-prometheusservice-controller components.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: server/meshmodel/aws-prometheusservice-controller/relationships/hierarchical_inventory.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{
+  "apiVersion": "core.meshery.io/v1alpha1",
+  "kind": "Hierarchical",
+  "metdata": {
+    "description": "Hierarchical"
```
Comment (leecalcote) [inline server/meshmodel/aws-prometheusservice-controller/relationships/hierarchical_inventory.json:5]: "description": "A hierarchical inventory relationship in which the configuration of (parent) component is patched with the configuration of other (child) component. Eg: The configuration of the EnvoyFilter (parent) component is patched with the configuration as received from WASMFilter (child) component."

File: server/meshmodel/aws-prometheusservice-controller/relationships/hierarchical_inventory.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    {
+      "allow": {
+        "from": [
+          {
+            "kind": "Pod",
+            "model": "Kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatorRef": [["type"], ["name"], ["namespace"]],
+              "description": "when a Pod interacts with an Adopted Resource, the Pod's configuration, like its type, name, and namespace, is updated based on the specifications from the Adopted Resource. This integration ensures smooth operation between Pods and Adopted Resources within the controller environment."
```
Comment (leecalcote) [inline server/meshmodel/aws-prometheusservice-controller/relationships/hierarchical_inventory.json:29]: Capitalize "when"

File: server/meshmodel/aws-prometheusservice-controller/relationships/hierarchical_inventory.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            "kind": "AdoptedResource",
+            "model": "aws-prometheusservice-controller",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatorRef": [
+                ["settings", "spec", "kubernetes", "kind"],
+                ["settings", "spec", "kubernetes", "metadata", "name"],
+                ["settings", "spec", "kubernetes", "metadata", "namespace"]
+              ],
+              "description": "The AdoptedResource component's Kubernetes-related specifications, such as kind, name, and namespace, are replaced to match the configurations of Pods. This enables seamless integration between AdoptedResources and Pods within the Kubernetes environment."
```
Comment (leecalcote) [inline server/meshmodel/aws-prometheusservice-controller/relationships/hierarchical_inventory.json:44]:
```suggestion
              "description": "Within the AWS Controllers for Kubernetes (ACK) project, the `AdoptedResource` serves a specific purpose: integrating existing, non-ACK-managed AWS resources into your Kubernetes cluster. `AdoptedResource` bridges the gap between pre-existing AWS resources and your Kubernetes cluster, fostering a more unified management experience. \n\nThis relationship allows the `AdoptedResource` configuration to be inherited from existing Kubernetes Pods."
```

File: server/meshmodel/aws-prometheusservice-controller/relationships/hierarchical_inventory.json (hunk 2 above)
Comment (leecalcote) [inline server/meshmodel/aws-prometheusservice-controller/relationships/hierarchical_inventory.json:29]:
```suggestion
              "description": "Within the AWS Controllers for Kubernetes (ACK) project, the `AdoptedResource` serves a specific purpose: integrating existing, non-ACK-managed AWS resources into your Kubernetes cluster. `AdoptedResource` bridges the gap between pre-existing AWS resources and your Kubernetes cluster, fostering a more unified management experience. \n\nThis relationship allows the `AdoptedResource` configuration to be inherited from existing Kubernetes Pods."
```

**Review by leecalcote — COMMENTED** [review]
I believe that this relationship is valuable. My understanding is that it might only be valuable to the extent that the user is in a brownfield situation, which is to say that they have already deployed their Pods (in this case). The behavior of the AdoptedResource (which is a resource pervasively available throughout each of the controller projects) is valuable to capture into a Meshery Relationship in-context of melding Visualizer (Operator) and Designer modes. The following is my understanding of the purpose and use of AdoptedResources. 

An AdoptedResource in the AWS Controllers for Kubernetes (ACK) project only applies to resources that have been deployed to a running cluster. It does not apply to resources defined solely in a manifest (not yet deployed).

Here's why:

1. Interaction with existing resources: The primary function of AdoptedResource is to manage existing AWS resources, not resources defined for future deployment. It achieves this by referencing the already provisioned resource and ensuring its state aligns with your configuration. Manifests, on the other hand, hold definitions for resources to be created, not existing ones.
1. Kubernetes API interaction: AdoptedResource interacts with the Kubernetes API to retrieve information about the existing resource and potentially manage its desired state. However, until a resource is deployed to the cluster and registered with the API, there's no information for AdoptedResource to interact with.

Therefore, while AdoptedResource offers a way to integrate existing resources with your Kubernetes cluster, it only takes effect once those resources are deployed and registered within the cluster.

I make note of this to help ensure that we understand when and where this relationship is useful to users, and how this moment in time (day 2 only), serves as a prerequisite that Meshery might need to enforce with the presences of the `id.pattern.meshery.io` annotation on the given existing Pod (in this case) resource.

@aabidsofi19 @MUzairS15 @theBeginner86

File: server/meshmodel/aws-prometheusservice-controller/relationships/hierarchical_inventory.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            }
+          }
+        ],
+        "to": [
+          {
+            "kind": "AdoptedResource",
+            "model": "aws-prometheusservice-controller",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatorRef": [
```
Comment (pandeyshubham03) [inline server/meshmodel/aws-prometheusservice-controller/relationships/hierarchical_inventory.json:39]: It should be `mutatedRef`, I believe!

**Pull request closed without merging**